File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.23-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_23() {
    let chars = ['a', 'b', 'c', 'd', 'e'];

    let row = 5;
    let column = 5;

    for (let y = row - 1; y >= 0; y--) {
        let element = y;

        for (let x = 0; x < column; x++) {
            if (element < column) {
                log(chars[element]);
                element = element + 1;
            } else {
                log(' ');
            }
        }
        console.log(' ', y);
    }

    console.log();

    return 0;
}

pattern_23();

File: python/00-Basics/00.03-build-up-logical-thinking/0.3.23-pattern.py
from utils.clear import clear

def pattern_23():
    row = 5
    column = 5

    chars = ['a', 'b', 'c', 'd', 'e']

    for i in range(row, 0, -1):
        counter = i - 1
        for j in range(column, 0, -1):
            if j >= i:
                print(chars[counter], end="")
                counter = counter + 1
            elif j <  i:
                print(" ", end="")

        print(" ", i)

clear()
pattern_23()
